### ISSUE TITLE
Make one-click OAuth external service connection possible on the backend

### ISSUE THREAD

**Issue by artemruts** [sourcegraph/sourcegraph#18805]
### Description

📃 [Slack conversation](https://sourcegraph.slack.com/archives/C01L84TMHMJ/p1614657491004900) for context.

To make one-click OAuth code host connections work we need to create a new record in `external_services` table when the backend captures the token from the auth provider.

I think it _may_  make sense to create a new `./api/[github|gitlab|*]/<some new name>` endpoint where I can send `GET` requests from the client.

The logic behind the new endpoint will be responsible for handling the OAuth dance (we can probably reuse exiting providers) and creating a new record in `external_services` table. 

`external_services` table will need to contain the `config` field with the correct JSON configuration for the given provider, where the `token` field is the captured OAuth token.

GitHub `config` example:
```json
{
  "url": "https://github.com",
  "token": "<captured token>",
  "orgs": []
}
```

### Acceptance Criteria (this + UI #18067)

To add a new OAuth code host connection:

1. On the client-side I can send a `GET` request to `./api/[github|gitlab|*]/<some new name>` endpoint and specify the `redirect` url with the query parameter
2. Once the backend OAuth logic completes client gets redirected to the url specified in `the redirect` query param

Once backend and frontend parts are complete the entire flow will look similar to the `Account security` page:

![tokens](https://user-images.githubusercontent.com/1319181/109728934-1ea61600-7b85-11eb-9114-668d977fa9d6.gif)

### Notes

Default configs for [GitHub](https://github.com/sourcegraph/sourcegraph/blob/7dbc725df1da7682199ade4a9f8a8b3c8bf5ae49/client/web/src/components/externalServices/externalServices.tsx#L508-L512) and [GitLab](https://github.com/sourcegraph/sourcegraph/blob/7dbc725df1da7682199ade4a9f8a8b3c8bf5ae49/client/web/src/components/externalServices/externalServices.tsx#L893-L899) external services

**Comment by unknwon** [comment]
@artemruts - I implemented a solution in https://github.com/sourcegraph/sourcegraph/pull/18926. PTAL and LMK if that works for you!

**Comment by artemruts** [comment]
> @artemruts - I implemented the solution for backend in #18926. PTAL and LMK if that works for you!

Thank you! Will test this out tonight 👍
